src/tab-list.js

```diff
@@ -55,6 +56,10 @@ export default class TabList extends React.Component {
     return (
       <ul className={className} onKeyUp={this.handleKeyPress} role="tablist">
         {React.Children.map(children, (child, index) => {
+          if (child.type !== Tab) {
```

Comment:
Does React not have a way to add restriction via the `propTypes`?

Reply:
It does via writing a custom proptype, which isn't super hard. My thought is that because things straight up break if a regular DOM element is a child, we want a restriction that works even if someone isn't heeding console warnings.

But, there's also an argument for enforcing through proptypes and assuming that people will use these components correctly.